task.go

```diff
@@ -301,6 +301,42 @@ func (et *ExecutingTask) registerOutput(name string, o Output) {
 	et.outputs[name] = o
 }
 
+type ExecutionStats struct {
+	TaskStats map[string]interface{}
+	NodeStats map[string]map[string]interface{}
+}
+
+func (et *ExecutingTask) ExecutionStats() (ExecutionStats, error) {
+	executionStats := ExecutionStats{
+		TaskStats: make(map[string]interface{}),
+		NodeStats: make(map[string]map[string]interface{}),
+	}
+
+	// Fill the task stats
+	executionStats.TaskStats["throughput"] = et.getThroughput()
+
+	// Fill the nodes stats
+	err := et.walk(func(node Node) error {
+		name := node.Name()
+
+		// Add the node stats
+		executionStats.NodeStats[name] = node.stats()
+
+		// Add collected and emitted
+		nodeStat := executionStats.NodeStats[name]
```

Comment:
Instead of indexing the map twice. And it reads cleaner so you know they are the same map.
```
nodeStat := node.stats()
executionStats.NodeStats[name]  = nodeStat
```